tracer/src/Datadog.Trace/Span.cs

```diff
@@ -77,7 +78,16 @@ internal Span(SpanContext context, DateTimeOffset? start, ITags tags)
         internal string ServiceName
         {
             get => Context.ServiceNameInternal;
-            set => Context.ServiceNameInternal = value;
+            set
+            {
+                if (!_baseServiceTagSet && !string.Equals(value, Context.ServiceNameInternal))
```

Comment:
We should do a culture-independent comparison here. Seeing as service names are ultimately normalized, I guess that we should probably ignore casing too? 🤔 

```suggestion
                if (!_baseServiceTagSet && !string.Equals(value, Context.ServiceNameInternal, StringComparison.OrdinalIgnoreCase))
```